From 6198a8b26c2cfca6b342347db8c3d8c64e2a5c82 Mon Sep 17 00:00:00 2001
From: Kosta Zertsekel <konszert@marvell.com>
Date: Wed, 3 Jul 2013 13:19:14 +0300
Subject: [PATCH 0756/1825] alp: eth complex: fix compilation when PPv2 is
 disabled in menuconfig

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit e109aaa9efc8d5e4a9190d005d101b17b40e06cb

	Ethernet Complex (for AvantaLP) is relevant only when PPv2 is compiled.
	When PPv2 is disabled in menuconfig/.config, Ethernet Complex should not
	be compiled and, hence, should not be initialized.

Signed-off-by: Kosta Zertsekel <konszert@marvell.com>
Change-Id: I12732a93af65b9ef418ec2a2a93e2d5b1051df12
Reviewed-on: http://vgitil04.il.marvell.com:8080/2407
Reviewed-by: Star_Automation <star@marvell.com>
Reviewed-by: Igor Patrik <igorp@marvell.com>
Tested-by: Igor Patrik <igorp@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/mach-avantalp/Makefile                    |    4 ++--
 .../avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.h    |    4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-avantalp/Makefile b/arch/arm/mach-avantalp/Makefile
index d834e3b..c0408eb 100644
--- a/arch/arm/mach-avantalp/Makefile
+++ b/arch/arm/mach-avantalp/Makefile
@@ -32,8 +32,7 @@ HAL_OBJS	= $(HAL_RTC_DIR)/mvRtc.o					\
 KW_FAM_OBJS	= $(BOARD_ENV_DIR)/mvBoardEnvSpec.o $(SOC_ENV_DIR)/mvCtrlEnvLib.o	\
 		  $(BOARD_ENV_DIR)/mvBoardEnvLib.o $(SOC_ENV_DIR)/mvCtrlEnvAddrDec.o 	\
 		  $(SOC_SYS_DIR)/mvAhbToMbus.o $(SOC_SYS_DIR)/mvCpuIf.o 		\
-		  $(SOC_CPU_DIR)/mvCpu.o $(SOC_DEVICE_DIR)/mvDevice.o			\
-		  $(SOC_ENV_DIR)/mvCtrlEthCompLib.o
+		  $(SOC_CPU_DIR)/mvCpu.o $(SOC_DEVICE_DIR)/mvDevice.o
 
 QD_OBJS		= $(HAL_QD_DIR)/src/driver/gtDrvConfig.o $(HAL_QD_DIR)/src/driver/gtDrvEvents.o \
                   $(HAL_QD_DIR)/src/driver/gtHwCntl.o $(HAL_QD_DIR)/src/platform/gtMiiSmiIf.o	\
@@ -68,6 +67,7 @@ avantalp-$(CONFIG_MV_INCLUDE_USB) 	+= $(HAL_USB_DIR)/mvUsb.o $(HAL_USB_DIR)/mvUs
 					   $(HAL_IF_DIR)/mvSysUsb.o
 avantalp-$(CONFIG_MV_INCLUDE_ETH_PHY)	+= $(HAL_ETHPHY_DIR)/mvEthPhy.o $(HAL_IF_DIR)/mvSysEthPhy.o
 
+avantalp-$(CONFIG_MV_ETH_PP2)		+= $(SOC_ENV_DIR)/mvCtrlEthCompLib.o
 obj-$(CONFIG_MV_ETH_PP2)                += $(LSP_PP2_DIR)/
 
 avantalp-$(CONFIG_MV_INCLUDE_CESA) 	+= $(HAL_CESA_DIR)/mvCesa.o					\
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.h
index 2205e07..198e2a4 100644
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.h
@@ -153,6 +153,10 @@ enum mvSwPortSrc {
 #define     ETHQPCS_DPLL_RESET_OFFSET			4
 #define     ETHQPCS_DPLL_RESET_MASK			(0x1 << ETHQPCS_DPLL_RESET_OFFSET)
 
+#ifdef CONFIG_MV_ETH_PP2
 MV_STATUS mvEthComplexInit(MV_U32 ethCompConfig);
+#else
+MV_STATUS mvEthComplexInit(MV_U32 ethCompConfig) { /* empty */ }
+#endif
 
 #endif /* __INCmvCtrlEthCompLibh */
-- 
1.7.5.4

